test/ipns.js

```diff
@@ -0,0 +1,328 @@
+        initOptions: { bits: 512 }
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        jsDaemon = node
+        jsDaemon.init(cb)
+      }),
+      (cb) => jsDaemon.start(cb),
+      (cb) => jsDaemon.api.id((err, res) => {
+        expect(err).to.not.exist()
+        jsId = res.id
+        cb()
+      }),
+      (cb) => jsDaemon.api.name.publish(ipfsRef, { resolve: false }, cb),
+      (cb) => {
+        jsDaemon.api.name.resolve(jsId, (err, res) => {
+          expect(err).to.not.exist()
+          expect(res).to.equal(ipfsRef)
+          cb()
+        })
+      },
+      (cb) => jsDaemon.stop(cb)
+      // (cb) => setTimeout(cb, 10500)
+    ], done)
+  })
+
+  it('should publish an ipns record to a go daemon and resolve it using the same go daemon', function (done) {
+    this.timeout(100 * 1000)
+
+    const dir = path.join(os.tmpdir(), hat())
+    const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+    let goDaemon1
+    let goDaemon2
+    let goId1
+
+    parallel([
+      (cb) => goDf.spawn({
+        repoPath: dir,
+        initOptions: { bits: 1024 }
+      }, cb),
+      (cb) => goDf.spawn({ initOptions: { bits: 1024 } }, cb)
+    ], (err, nodes) => {
+      expect(err).to.not.exist()
+
+      goDaemon1 = nodes[0]
+      goDaemon2 = nodes[1]
+
+      parallel([
+        (cb) => goDaemon1.start(cb),
+        (cb) => goDaemon2.start(cb)
+      ], (err, nodes) => {
+        expect(err).to.not.exist()
+
+        series([
+          (cb) => goDaemon1.api.id((err, res) => {
+            expect(err).to.not.exist()
+            goId1 = res
+            cb()
+          }),
+          (cb) => goDaemon1.api.name.publish(ipfsRef, { resolve: false }, cb),
+          (cb) => {
+            goDaemon1.api.name.resolve(goId1, { resolve: false }, (err, res) => {
+              expect(err).to.not.exist()
+              expect(res).to.equal(ipfsRef)
+              cb()
+            })
+          },
+          (cb) => goDaemon1.stop(cb),
+          (cb) => goDaemon2.stop(cb)
+        ], done)
+      })
+    })
+
+    // Error: failed to find any peer in table
+    /*
+    series([
+      (cb) => goDf.spawn({
+        repoPath: dir,
+        initOptions: { bits: 1024 },
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        goDaemon1 = node
+        cb()
+      }),
+      (cb) => goDaemon1.start(cb),
+      (cb) => goDaemon1.api.id((err, res) => {
+        expect(err).to.not.exist()
+        goId1 = res
+        cb()
+      }),
+      (cb) => goDaemon1.api.name.publish(ipfsRef, { resolve: false }, cb),
+      (cb) => {
+        goDaemon1.api.name.resolve(jsId, (err, res) => {
+          expect(err).to.not.exist()
+          expect(res).to.equal(ipfsRef)
+          cb()
+        })
+      },
+      (cb) => goDaemon1.stop(cb),
+    ], done) */
+  })
+
+  it('should publish an ipns record to a js daemon and resolve it using a go daemon through the reuse of the same repo', function (done) {
```

Comment:
There is a problem with the repo in this test. I have also added a test for simulating this in the repo tests.

Basically, If we try to reuse to same repo previously used in JS (after running init) using a non-disposable daemon, the start fails with `Error: required Datastore.Spec entry missing from config file`. I tried to add the datastore config to the spawn method (in the repo test, but I am not sure if it is correct) but I couldn’t figure out the problem here.